rules/library.bzl

```diff
@@ -958,15 +962,12 @@ def apple_library(
     )
 
     if swift_sources:
-        swift_kwargs = kwargs
-
-        # To be backward compatible with rules_apple 2.x
-        if plugins:
-            swift_kwargs = dicts.add(swift_kwargs, {"plugins": plugins})
-
-        # Allow consumers to opt-in to https://github.com/bazelbuild/rules_swift/pull/1162
-        if always_include_developer_search_paths:
-            swift_kwargs = dicts.add(swift_kwargs, {"always_include_developer_search_paths": always_include_developer_search_paths})
+        # Collect the swift_library related kwargs, these are typically only set when provided to allow
+        # for wider compatibility with rule_swift versions.
+        swift_kwargs = dicts.add(
+            kwargs,
+            {arg: kwargs.pop(arg) for arg in _SWIFT_LIBRARY_KWARGS if arg in kwargs},
```

Comment:
Have you verified if pop-ing this only here instead of up around line 581 causes problems? IIUC if we don't pop these early it means we'll pass these flags to the [modulemap_generator](https://github.com/bazel-ios/rules_ios/blob/eab038c4028156f7b3825ae9bf2a28675ac8ccb8/rules/library.bzl#L836) correct?

Reply:
Good point, updated this to collect/pop the swift related kwargs earlier and then add them to kwargs right before the library is created.